frontend/src/app/services/session.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+import { Observable, BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+const URL = 'http://localhost:8080/';
+const httpOptions = {
+  headers: new HttpHeaders({
+    'Authorization': 'Basic ' + btoa("fooClientIdPassword:secret"),
+    'Content-type': 'application/x-www-form-urlencoded'
+  })
+};
+
+@Injectable()
+
+export class SessionService {
+
+  private _userData$: BehaviorSubject<any> = new BehaviorSubject(undefined);
+  private _isLoggedIn$: BehaviorSubject<boolean> = new BehaviorSubject(this.hasToken());
+  private body : HttpParams;
+
+  
+  constructor(private http : HttpClient) { }
+
+
+  public login(username, password){
```

Comment:
You need the rework this part of the code as it returns no response to the caller component whether the login was successful or not. There are multiple ways to achieve this.
1. you only pipe to the data stream of the HTTP response and return the Observable so any component could subscribe to it (error handling as the error Input in the component comes from nowhere

```suggestion
  public login(username, password): Observable<any> {
    this.body = new HttpParams()
      .set('username', username)
      .set('password', password)
      .set('grant_type', 'password');

    return this.http.post(URL + 'oauth/token', this.body, httpOptions)
      .pipe(tap((data: any) => {
        localStorage.setItem('token', data.access_token);
        this._isLoggedIn$.next(true);
      }));
  }
```